src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -140,75 +161,28 @@ private enum Flags : byte
-                }
-
-                bool isGetterOnly = hasGetAccessor && !hasSetAccessor;
-
-                if (isAutoPropertyWithGetSyntax && !IsStatic && !isGetterOnly && !isInitOnly)
-                {
-                    if (ContainingType.IsReadOnly)
-                    {
-                        diagnostics.Add(ErrorCode.ERR_AutoPropsInRoStruct, location);
-                    }
-                    else if (HasReadOnlyModifier)
-                    {
-                        diagnostics.Add(ErrorCode.ERR_AutoPropertyWithSetterCantBeReadOnly, location, this);
-                    }
-                }
-
-                if (isAutoPropertyWithGetSyntax || hasInitializer)
-                {
-                    if (isAutoPropertyWithGetSyntax)
-                    {
-                        //issue a diagnostic if the compiler generated attribute ctor is not found.
-                        Binder.ReportUseSiteDiagnosticForSynthesizedAttribute(DeclaringCompilation,
-                        WellKnownMember.System_Runtime_CompilerServices_CompilerGeneratedAttribute__ctor, diagnostics, syntax: syntax);
-
-                        if (this._refKind != RefKind.None && !_containingType.IsInterface)
-                        {
-                            diagnostics.Add(ErrorCode.ERR_AutoPropertyCannotBeRefReturning, location, this);
-                        }
-                    }
-
-                    string fieldName = GeneratedNames.MakeBackingFieldName(_sourceName);
-                    BackingField = new SynthesizedBackingFieldSymbol(this,
-                                                                          fieldName,
-                                                                          isReadOnly: isGetterOnly || isInitOnly,
-                                                                          this.IsStatic,
-                                                                          hasInitializer);
-                }
-
-                if (isAutoProperty)
-                {
-                    Binder.CheckFeatureAvailability(
-                        syntax,
-                        isGetterOnly ? MessageID.IDS_FeatureReadonlyAutoImplementedProperties : MessageID.IDS_FeatureAutoImplementedProperties,
-                        diagnostics,
-                        location);
-                }
+                string fieldName = GeneratedNames.MakeBackingFieldName(_sourceName);
+                BackingField = new SynthesizedBackingFieldSymbol(this,
+                                                                      fieldName,
+                                                                      isReadOnly: (hasGetAccessor && !hasSetAccessor) || isInitOnly,
+                                                                      this.IsStatic,
+                                                                      hasInitializer);
             }
+        }
 
+        private void EnsureSignature(DiagnosticBag diagnostics)
```

Comment:
> Believe this can only legally called from EnsureSignature() correct? If so did you consider making this a local function there to prevent bad calls? 

To simplify review process I tried to keep original code as close to the original place as possible and tried to make as little as possible changes around it. Since the method is private, I do not see significant advantage in using a local function at this point.

---
In reply to: [526219649](https://github.com/dotnet/roslyn/pull/49360#discussion_r526219649) [](ancestors = 526219649)